### PR TITLE
No need to apply `to_sym` on symbol

### PR DESCRIPTION
**Pull request by akshay-vishnoi**

### CODE REVIEW

File: actioncable/lib/action_cable/channel/base.rb [hunk 1]
```diff
@@ -230,7 +230,7 @@ def subscribe_to_channel
 
 
         def extract_action(data)
-          (data['action'].presence || :receive).to_sym
+          data['action'].presence.to_sym || :receive
```
Comment (kaspth) [inline actioncable/lib/action_cable/channel/base.rb]: Won't work as `presence` can return nil and:

```ruby
nil.to_sym
# raises NoMethodError: undefined method `to_sym' for nil:NilClass
```

Comment (akshay-vishnoi) [inline actioncable/lib/action_cable/channel/base.rb]: Should we use try here or original was fine?

**Comment by kaspth** [comment]
Thanks! Unfortunately this can now raise `NoMethodError`s. I don't think it's worth `if`-ing around that to avoid it. :grin:

File: actioncable/lib/action_cable/channel/base.rb (hunk 1 above)
Comment (kaspth) [inline actioncable/lib/action_cable/channel/base.rb]: Benchmarked it. Doesn't look like calling `to_sym` on a symbol is a potential bottleneck:

```ruby
require 'benchmark/ips'

Benchmark.ips do |x|
 x.report('to_sym') { :receive.to_sym }
end
```

Got:

```
Calculating -------------------------------------
              to_sym   131.252k i/100ms
-------------------------------------------------
              to_sym      7.973M (± 7.1%) i/s -     39.769M
```

Original seems fine, thanks! :grin:

Comment (akshay-vishnoi) [inline actioncable/lib/action_cable/channel/base.rb]: Cool. Thanks for feedback

Comment (kaspth) [inline actioncable/lib/action_cable/channel/base.rb]: No problem :metal:

Comment (akshay-vishnoi) [inline actioncable/lib/action_cable/channel/base.rb]: I think that in this case, we should add a test case to test `perform_action` with empty data. According to the code `:recieve` action should be dispatched, hence we should expect this in the test case. WDYT? Should I add the test case for that?

Comment (kaspth) [inline actioncable/lib/action_cable/channel/base.rb]: Sure, feel free to take a stab at it :+1:
